src/rgw/rgw_op.cc

```diff
@@ -3893,6 +3903,9 @@ void RGWPutObj::execute(optional_yield y)
       } else {
         compressor.emplace(s->cct, plugin, filter);
         filter = &*compressor;
+        if (compression_hint == "incompressible_only_compressed") {
+          obj_ctx.alloc_hint_flags |= librados::ALLOC_HINT_FLAG_INCOMPRESSIBLE;
+        }
```

Comment:
when rgw is itself doing compression, i tend to think we should _always_ send this incompressible hint

same goes for encryption too, right?